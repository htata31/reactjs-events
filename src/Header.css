
.navbar-title {
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: 1.5em;
    text-transform: capitalize;
  }
.navbar-light .navbar-text{
    color:white;
    padding: 0;
  }
  .navbar-light .navbar-brand{
      padding:0;
  }