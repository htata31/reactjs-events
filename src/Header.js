import React, {Component} from 'react';
import './bootstrap.min.css';
import './Header.css'
import { Navbar } from 'react-bootstrap';


class Header extends Component {
    render(){
        return (
            <Navbar sticky="top" id="navbar" className="bg-dark justify-content-between">
                <Navbar.Brand href='https://www.call-em-all.com/'><img alt='Call-Em-All Logo' id='call-em-all-logo' src='Images/Call-Em-All-logo.png'/></Navbar.Brand>
                <Navbar.Text className='navbar-title'> {this.props.title}
                </Navbar.Text>
                <Navbar.Brand href='https://www.meetup.com/'><img alt='Meetup Logo' id='meetup-logo' src='Images/meetup_logo.png'/></Navbar.Brand>
            </Navbar>
        );
    }
  }
  
  export default Header;