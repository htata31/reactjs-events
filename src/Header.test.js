import enzyme, { shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import toJson from 'enzyme-to-json';
import Header from './Header'


enzyme.configure({ adapter: new enzymeAdapterReact16(), });
const header_title = 'React Bootstrap Meetup Application';

function getHeaderShallowWrapper() {
    return shallow(
      <Header title = {header_title} />
    );
  }

  describe('The Header component', () => {
    test('Taking a snapshot of the header', () => {
      const headerWrapper = getHeaderShallowWrapper().find('#navbar');
  
      expect(toJson(headerWrapper)).toMatchSnapshot();
    });
  
    test('render meetup logo', () => {
        const meetLogo = getHeaderShallowWrapper().find('#meetup-logo');
    
        expect(meetLogo.exists()).toBe(true);
        expect(meetLogo.children().length).toBe(0);
      });
    
      test('render Call em all logo', () => {
        const callEmAll = getHeaderShallowWrapper().find('#call-em-all-logo');
    
        expect(callEmAll.exists()).toBe(true);
        expect(callEmAll.children().length).toBe(0);
      });

      test("renders an meetup logo image", () => {
        const logo = getHeaderShallowWrapper();

        expect(logo.find("#meetup-logo").prop("src")).toEqual('Images/meetup_logo.png');
      });

      test("renders an call em all logo image", () => {
        const logo = getHeaderShallowWrapper();

        expect(logo.find("#call-em-all-logo").prop("src")).toEqual('Images/Call-Em-All-logo.png');
     });
  
  });