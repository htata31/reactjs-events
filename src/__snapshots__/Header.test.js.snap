// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component Taking a snapshot of the header 1`] = `
<Navbar
  className="bg-dark justify-content-between"
  collapseOnSelect={false}
  expand={true}
  id="navbar"
  sticky="top"
  variant="light"
>
  <NavbarBrand
    href="https://www.call-em-all.com/"
  >
    <img
      alt="Call-Em-All Logo"
      id="call-em-all-logo"
      src="Images/Call-Em-All-logo.png"
    />
  </NavbarBrand>
  <NavbarText
    className="navbar-title"
  >
     
    React Bootstrap Meetup Application
  </NavbarText>
  <NavbarBrand
    href="https://www.meetup.com/"
  >
    <img
      alt="Meetup Logo"
      id="meetup-logo"
      src="Images/meetup_logo.png"
    />
  </NavbarBrand>
</Navbar>
`;
